proto/p4/v1/p4runtime.proto

```diff
@@ -210,6 +210,18 @@ message TableEntry {
   IdleTimeout time_since_last_hit = 10;
   // Arbitrary metadata from the controller that is opaque to the target.
   bytes metadata = 11;
+  // const is true only for entries that cannot be deleted or
+  // modified, e.g. any of the following:
+  // + Any entry read from a table declared with `const entries`
+  // + The default entry read from a table declared with `const default_action`
```

Comment:
I didn't realize that default actions generate entries? Is that true for all targets and standardized somewhere, or a target-specific thing?

Reply:
Is it perhaps clearer if I said "The default action read from a table" instead of "The default entry read from a table"?

The actual hardware resource utilization of a default action in a P4 table is potentially a little target-specific, e.g. for a ternary table, I can imagine implementations that use a TCAM entry that is completely wildcarded with lowest match priority to implement a default_action, whereas another implementation might have a separate place to store a default action outside of the TCAM completely.  Typically it only makes a difference of one hardware entry used vs. not in implementations I am familiar with.

Reply:
Every P4 table always has a default action.  If it isn't explicitly specified in P4 source code, it defaults to NoAction

Reply:
But at least on our switches, there is no table entry that I can read back via P4Runtime that implements the default action.

Reply:
P4Runtime API read requests can read either one (or all) "hit entries", or the default action of a table, never both in the same read request.  See mentions of the field `is_default_action` throughout the P4Runtime spec.

What happens if you send a ReadRequest to one of your switches with `is_default_action` true?

Reply:
I may not have been clear above, but note that what P4Runtime API defines is that there are 0 or more entries in a table, which if any of them are matched the device must behave as if table_name.apply().hit is true, and if none of them match, the device must behave as if table_name.apply().hit is false.  In the latter case, the default_action of the table must be executed.

As an implementation detail, one correct way to implement that behavior for a hardware TCAM is to have a separate hardware TCAM hit entry with wildcard value/mask, and an action equal to the currently configured default_action.  A correct P4Runtime API implementation for such a hardware implementation should _never_ return the hardware entry representing the default_action from a P4Runtime API read request that is reading entries installed in the table, because that hardware entry is an implementation detail for how that device chose to implement default_action behavior.

Reply:
Turns out I had some catching up to do on "default entries" and `is_default_action`. All makes sense now, thanks, and sorry if I didn't make sense.

> What happens if you send a ReadRequest to one of your switches with is_default_action true?

Our switches currently only support wild card reads. And I just double checked that wildcard reads do not include the default entry, which is probably why I wasn't super aware they were a thing thus far. So all is good.